### ISSUE TITLE
HAML

### ISSUE THREAD

**Issue by waynehoover** [zed-industries/extensions#179]
### Check for existing issues

- [X] Completed

### Language

HAML

### Misc notes

This isn't a language per se; it's a markup language. It would be nice for us Rails developers who work in HAML frequently to have some syntax coloring.

https://github.com/haml/haml

This was also requested as a comment in this issue for a similar markup language, slim: https://github.com/zed-industries/extensions/issues/140

**Comment by arbales** [comment]
This is my #1 barrier / frustration.

**Comment by CyberFerret** [comment]
This is one of the main things holding me back from converting from VS Code to Zed - No syntax highlighting for my HAML files is a big step backwards in productivity.  The Ruby syntax highlighting is great, just hoping to see it incorporate HAML files as well!

**Comment by Teyler7** [comment]
I'd be happy to help the Zed team on this if needed 👍

**Comment by czj** [comment]
There has been an awesome of the Slim templating language, which made us switch 100% to Zed from Sublime & VS Code. I'm confident HAML porting would be quite similar.

Teyler7 made you could open a PR similar to https://github.com/zed-industries/extensions/pull/1007 but for HAML to ?

**Comment by davidcornu** [comment]
I have something very basic working over at https://github.com/davidcornu/zed-haml based on https://github.com/vitallium/tree-sitter-haml.

![CleanShot 2024-09-13 at 13 37 02@2x](https://github.com/user-attachments/assets/efa8dce2-bafb-4d3f-af19-0fbc63873f75)

**Comment by avdept** [comment]
> https://github.com/davidcornu/zed-haml

Great work mate, installed it myself. At first glance everything works as intended

**Comment by CyberFerret** [comment]
Fantastic work @davidcornu 👏🏽 I installed it locally (Downloaded your repo, then installed via the 'Developer Extensions' button on the top right of the Extensions view in Zed) and getting some great Syntax highlighting in HAML files now.

The only thing that would make it better would be if the colours of Object names within inline code in the HAML was a different colour like it is in the Ruby code syntax highlighting, but what is there now makes HAML editing SO much better, so thanks again.

**Comment by davidcornu** [comment]
Yeah the parser I'm using is still a work in progress. I'm very new to tree sitter but will see if I can contribute some improvements when I have some spare time.

**Comment by firecall** [comment]
> https://github.com/davidcornu/zed-haml

Amazing, thankyou!

Working great!

HAML support has been holding me back from using Zed!

**Comment by richorton** [comment]
Thanks, works great 🙏🏻

**Comment by davidcornu** [comment]
![CleanShot 2024-10-15 at 16 54 36@2x](https://github.com/user-attachments/assets/15841cbc-d9fd-4ef7-9ca7-6d194d1d4436)

https://github.com/davidcornu/zed-haml has been published as an extension https://github.com/zed-industries/extensions/pull/1523 so anyone using it in developer mode can switch to the regular installation path.

**Comment by JosephTLyons** [comment]
<img width="1142" alt="Image" src="https://github.com/user-attachments/assets/5fa2b725-3156-4945-addb-05f9160916b0" />
